src/System.Net.Sockets/src/System/Net/Sockets/BaseOverlappedAsyncResult.Windows.cs

```diff
@@ -95,14 +95,17 @@ private static unsafe void CompletionPortCallback(uint errorCode, uint numBytes,
                     }
                     else
                     {
+                        bool addRefedNativeOverlapped = false;
                         try
                         {
+                            asyncResult._nativeOverlapped.DangerousAddRef(ref addRefedNativeOverlapped);
```

Comment:
Yeah, I guess my question is really, is this necessary?  What does it protect against?